common/src/main/java/com/zegelin/prometheus/netty/Server.java

```diff
@@ -62,27 +70,44 @@ public static void start(final List<InetSocketAddress> listenAddresses, final Ha
                 .channel(NioServerSocketChannel.class)
                 .childHandler(new ChildInitializer(harvester, helpExposition));
 
-        final List<Channel> channels;
         {
             final ImmutableList.Builder<Channel> builder = ImmutableList.builder();
 
             for (final InetSocketAddress listenAddress : listenAddresses) {
-                final Channel channel = bootstrap.bind(listenAddress).sync().channel();
-
-                builder.add(channel);
+                builder.add(bootstrap.bind(listenAddress).sync().channel());
             }
 
-            channels = builder.build();
+            Server.channels = builder.build();
         }
 
         if (logger.isInfoEnabled()) {
-            logger.info("cassandra-exporter started. Listening on {}", Joiner.on(", ").join(
+            logger.info("Cassandra Exporter Agent Server has started. Listening on {}", Joiner.on(", ").join(
                     listenAddresses.stream()
                             .map(a -> String.format("http://%s:%d", a.getHostString(), a.getPort()))
                             .iterator()
             ));
         }
+    }
+
+    public static void stop() {
```

Comment:
Does explicitly stopping the server change the behaviour or fix anything?

Since the HTTP server threads are marked as daemon, they should automatically be stopped when the application exits. Not that having the functionality to explicitly stop the server isn't a good idea, and is certainly more explicit and clear in terms of behaviour.

Reply:
It does not seem to change any behavior and you are right about daemon threads but on the other hand, if you want to start / stop this agent / server programmatically (for whatever reason, one of them is e.g. testing (if that is even somehow possible, I havent explored this area yet)), isn't it true one would want to control starting and stopping of this explicitly?

Ideally I would go so far that that I would make some JMX bean via which you can turn this on / off so JAR would be still attached there but it would not collect any metrics to be exported. The only way how to turn this off, as of now, is stopping Cassandra and removing that JAR from the class path.

Reply:
For the agent version, as it's designed currently, requires the agent to install its `MBeanServerInterceptor` as early as possible in the JVM startup lifecycle so it's able to intercept the registration of the MBeans created by Cassandra.

So while it's possible to start and stop the HTTP server, it would require a bit of a rework of the internals to be able to start & stop the whole thing.